### DISCUSSION TITLE
Full GDB Console

### DISCUSSION THREAD

**Discussion by The-real-Thomas-A-Anderson** [eclipse-embed-cdt/eclipse-plugins#558] [Debug Plug-ins Q&A] [answered]
Hi, 

Is there a way to enable Full GDB Console? To have some features like command-completion, command-history, etc. in gdb console.
I stumbled upon this: https://wiki.eclipse.org/CDT/User/NewIn92#Full_GDB_Console

I managed to get the "Debugger Console". There is a slightly non-standard behavior, i.e. missing (gdb) prefix at the command line.

And most important: I am on windows. The link above talks about linux. 
Quote: "CDT 9.2, when used on Linux with GDB 7.12, provides a full-featured GDB command-line console in Eclipse. This console is accessible through the new Debugger Console view.

My gdb version:
> show version
> GNU gdb (xPack GNU Arm Embedded GCC x86_64) 11.2.90.20220202-git

Many thanks in advance

**Comment by ilg-ul** [comment]
As far as I remember, Embedded CDT does nothing special regarding this console, the behaviour should come from CDT.

**Comment by jonahgraham** [comment, marked as answer]
> And most important: I am on windows. The link above talks about linux.

Yes, this feature only works on Linux today as it relies on OS functionality that does not exist on Windows.

There is a feature request tracking this on the CDT GitHub repo here: https://github.com/eclipse-cdt/cdt/issues/362

**Reply by The-real-Thomas-A-Anderson** [reply]
Thanks for pointing this out.
Not the first time I stumble upon this issue...
